### DISCUSSION TITLE
Dark shell

### DISCUSSION THREAD

**Discussion by trya2l** [Kidel/Kathara#286] [Q&A] [answered]
Hi,

I would like to know if there is a good way to turn back the shell in dark theme ?

![screenshot-24-04-20-15-56-02](https://github.com/KatharaFramework/Kathara/assets/79646756/312ea0e2-46b0-4e91-b07b-7723e963c400)
This is a bit harsh on my eyes...

But guys, kathara is quite awesome, the tool is so fun that it healed some network trauma I've had during school :)

**Comment by tcaiazzi** [comment, marked as answer]
Dear @trya2l, 

Thanks for reaching out! We are super happy that you like Kathará.

What terminal emulator are you using? 

The terminal theme is not related to Kathará, it depends on the theme you are using with your terminal emulator. 

If you are using `xterm`, as by default, you need to create a `.Xresources` file containing your customization. 
For example: 
```
xterm*dynamicColors: true
xterm*background: black
xterm*foreground: grey
xterm*saveLines: 35000
```
Then, you need to update the settings before they get effective.
```
xrdb ~/.Xresources 
```
![immagine](https://github.com/KatharaFramework/Kathara/assets/26346836/608c8a8b-ed2a-433f-9848-0548aa02c1d6)

You can find more info on how to configure `xterm` [here](https://wiki.archlinux.org/title/Xterm).

Please let me know if this helps, or if I can help you further! 😄

**Reply by trya2l** [reply]
Hi, thanks for your answer.

Yes, I saw afterwards that this was manageable using `kathara settings` 

However, this brings another question about what terminals are supported.

I can not seem to be able to use things like `alacritty` and `kitty`, but `gnome-terminal` is supported.

Do you know something about this?

This seems to be related to `Wayland` and `dbus`

**Comment by tcaiazzi** [comment]
Hi @trya2l, 

To include different terminal emulators, it is needed to write a specific connector for them, unless they have the same CLI of `xterm`. In this case, you can specify your custom shell in the settings. 

For what regards `gnome-terminal`, we decided to include it since it is quite common and better supports high resolutions than `xterm` (which needs proper customization to finely work on 2K/4K displays). Moreover, it has a super simple interface, very similar to the one of `xterm`, so it was easy to implement.

The `TMUX` connector was added as a PR (#98) ❤️ 

Do you have any specific reason for adding `alacritty` or `kitty`?

**Reply by trya2l** [reply]
Hi,

Thanks, it makes sense. I was simply curious about the fact that every terminal didn't work out of the box.

But the TMUX one intrigues me. I thought TMUX needed a terminal to run. I tried it, but nothing launched. What am I missing ? (and sorry for dragging this question a bit)

**Reply by tcaiazzi** [reply]
Hi @trya2l! 

Don't worry, I'm happy to explain, especially here on GitHub, so that this can help other people as well 😄 

When you run the `kathara lstart` command, it creates a TMUX session. You need to attach your terminal to the TMUX session. 
To do so, run the following command: 
```
tmux attach
```
This command will attach you to the last created session.

![immagine](https://github.com/KatharaFramework/Kathara/assets/26346836/0a7f08d9-0640-4c8d-956c-778ddf87dcdf)

In the footer, you can see devices' names associated with a number. To navigate among different devices' shell, you can use: `CTRL + B` and then press the number that corresponds to the desired device:
- e.g.: `CTRL + B -> 1`, to log into `rootdns` in the screen above. 

Here is a link to a TMUX cheat sheet: https://tmuxcheatsheet.com/.
For a complete reference of TMUX commands, see the man pages.

**Reply by trya2l** [reply]
Cool, thanks you then. So that's the purpose of the TMUX option. It sounds interesting. I'll check this out.

Thanks again 😁
